gateway/ln-gateway/src/ng/tests.rs

```diff
@@ -124,7 +74,8 @@ pub fn sha256(data: &[u8]) -> sha256::Hash {
 #[tokio::test(flavor = "multi_thread")]
 async fn test_gateway_client_pay_valid_invoice() -> anyhow::Result<()> {
     gateway_test(
-        |_, other_lightning_client, _, user_client, gateway| async move {
+        |gateway, other_lightning_client, fed, user_client| async move {
+            let gateway = gateway.remove_client(&fed).await;
```

Comment:
@m1sterc001guy we remove the client from the gateway so you can manually manipulate the gateway.  However, we could also not do this and let this be more of an E2E test that will test the gateway RPC logic.

Reply:
ah ok I think I understand. So `GatewayTest` here will do actual RPCs to the webserver. I think that works, maybe we could just rename `gateway` here to be `gateway_client` to make that more clear